parser-typechecker/src/Unison/Codebase/SqliteCodebase.hs

```diff
@@ -183,93 +158,27 @@ initSchemaIfNotExist path = liftIO do
   unlessM (doesDirectoryExist $ makeCodebaseDirPath path) $
     createDirectoryIfMissing True (makeCodebaseDirPath path)
   unlessM (doesFileExist $ makeCodebasePath path) $
-    withConnection "initSchemaIfNotExist" path $ runReaderT Q.createSchema
+    withConnection "initSchemaIfNotExist" path \conn ->
+      Sqlite.runTransaction conn Q.createSchema
```

Comment:
@mitchellwrosen what influences the decision to pick `runWriteTransaction` in some spots and `runTransaction` here? I suspect we probably want to use the appropriate `read/write` version in cases where we know the type of transaction for sure?

```suggestion
      Sqlite.runWriteTransaction conn Q.createSchema
```